admin/create-theme/theme-readme.php

```diff
@@ -137,123 +152,154 @@ static function copyright_section( $new_copyright_section, $original_theme_credi
+			$readme_content = rtrim( $readme_content );
 
-		$recommended_plugins_section = $section_start . "\n" . $recommended_plugins . "\n";
+			// Ensure two newlines before appending new section
+			if ( ! empty( $readme_content ) ) {
+				$readme_content .= "\n\n\n";
+			}
 
-		if ( $updated_readme ) {
-			return $updated_readme . $recommended_plugins_section;
+			// Append new section if not found
+			$updated_content = $readme_content . $replacement;
 		}
 
-		return $recommended_plugins_section;
+		return $updated_content;
 	}
 
 	/**
-	 * Update current readme.txt file, rather than building a new one.
+	 * Adds or updates a property in the readme content.
 	 *
-	 * @return string
+	 * @param string $prop_name The name of the property.
+	 * @param string $prop_value The value of the property.
+	 * @param string $readme_content The content of the readme file.
+	 * @return string The updated readme content.
 	 */
-	public static function update_readme_txt( $theme ) {
-		$description         = $theme['description'];
-		$author              = $theme['author'];
-		$wp_version          = get_bloginfo( 'version' );
-		$image_credits       = $theme['image_credits'] ?? '';
-		$recommended_plugins = $theme['recommended_plugins'] ?? '';
-		$updated_readme      = '';
-		$current_readme      = get_stylesheet_directory() . '/readme.txt' ?? '';
-		$readme_content      = file_exists( $current_readme ) ? file_get_contents( $current_readme ) : '';
-
-		if ( ! $readme_content ) {
-			return;
-		}
-
-		$updated_readme = $readme_content;
-
-		// Update description.
-		if ( $description ) {
-			$pattern = '/(== Description ==)(.*?)(\n\n=|$)/s';
-			preg_match_all( $pattern, $updated_readme, $matches );
-			$current_description = $matches[0][0];
-			$updated_readme      = str_replace( $current_description, "== Description ==\n\n{$description}\n\n=", $updated_readme );
+	private static function add_or_update_prop( $prop_name, $prop_value, $readme_content ) {
+		if ( empty( $prop_value ) ) {
+			return $readme_content;
 		}
+		$pattern = '/(' . preg_quote( $prop_name, '/' ) . ')(.*?)(\n|$)/s';
+		preg_match_all( $pattern, $readme_content, $matches );
+		$current_uri    = $matches[0][0];
+		$updated_readme = str_replace( $current_uri, "{$prop_name}: {$prop_value}\n", $readme_content );
+		return $updated_readme;
+	}
 
-		// Update Author/Contributors.
-		if ( $author ) {
-			$pattern = '/(Contributors:)(.*?)(\n|$)/s';
-			preg_match_all( $pattern, $updated_readme, $matches );
-			$current_uri    = $matches[0][0];
-			$updated_readme = str_replace( $current_uri, "Contributors: {$author}\n", $updated_readme );
-		}
+	/**
+	 * Write the readme.txt file.
+	 *
+	 * @param string $content The content to write.
+	 * @return bool True if the content was written, false otherwise.
+	 */
+	public static function write( $content ) {
+		$path = self::file_path();
+		return (bool) file_put_contents( $path, $content );
+	}
```

Comment:
This function is used only in a single place (beyond the tests) and that could just be replace with a call to file_put_contents().  I don't believe this adds value, just complication.

Reply:
Removed here: https://github.com/WordPress/create-block-theme/pull/626/commits/2d4cedbab2ab13e69a1d475b7879ebdce5b4e374

Reply:
🧼